src/genie/libs/parser/iosxr/show_interface.py

```diff
@@ -3032,4 +3032,72 @@ def cli(self, interface=None, output=None):
         else:
             out = output
 
-        return super().cli(output=out)
\ No newline at end of file
+        return super().cli(output=out)
+
+
+#############################################################################
+# Parser For show interface summary
+#############################################################################
+
+class ShowInterfaceSummarySchema(MetaParser):
+    """schema for show interface summary
+    """
+
+    schema = {
+        'interface_types': {
+            Any(): {
+                'total': str,
+                'up': str,
+                'down': str,
+                'admin_down': str,
+            }
+        }
+    }
+
+
+class ShowInterfaceSummary(ShowInterfaceSummarySchema):
+    """parser for show interface summary
+    """
+
+    cli_command = ['show interface summary']
+
+    def cli(self, interface="", output=None):
+        if output is None:
+            if interface:
+                cmd = self.cli_command[1].format(interface=interface)
+            else:
+                cmd = self.cli_command[0]
+            out = self.device.execute(cmd)
+        else:
+            out = output
+
+        interface_dict = {}
+        ret_dict = {}
+
+        # <INTERFACE_TYPE_NAME> 60 20 0 1
+        p1 = re.compile(r'^(?P<type>\w+\s?\w+)\s+(?P<total>\d+)\s+(?P<up>\d+)\s+(?P<down>\d+)\s+(?P<admin_down>\d+)$')
+
+        for line in out.splitlines():
+            line = line.strip()
+
+            res = p1.match(line)
+            if res:
+                if 'interface_types' not in ret_dict:
+                    interface_dict = ret_dict.setdefault('interface_types', {})
```

Comment:
No need for an if guard, set default is no clobber.

```suggestion
                interface_dict = ret_dict.setdefault('interface_types', {})
```